### PR TITLE
Adding an "upload" method to the Media endpoint

### PR DESCRIPTION
**Pull request by boboudreau**

### CODE REVIEW

File: specs/Endpoint/media.spec.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+
+describe(Media::class, function () {
+    beforeEach(function () {
+        $this->wpClient = $this->getProphet()->prophesize(WpClient::class);
+        $this->mediaEndpoint = new Media($this->wpClient->reveal());
+    });
+
+    xit('should throw a Runtime Exception if a remote file is not found', function () {
```
Comment (rlehrke) [inline specs/Endpoint/media.spec.php:18]: Do we not like this test right now?

File: specs/Endpoint/media.spec.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    it('should throw a Runtime Exception if the file is not found', function () {
+        $filename = realpath(dirname('../')) . '/file-does-not-exist.txt';
+        set_error_handler(function (int $errno, string $errstr, string $errfile) {
+            expect($errno)->to->equal(E_WARNING);
+            expect(strpos($errstr, ' No such file or directory'))->to->not->equal(false);
+        });
+        try {
+            $response = $this->mediaEndpoint->upload($filename);
+        } catch(Exception $e) {
+            expect($e)->to->be->instanceof(RuntimeException::class);
```
Comment (rlehrke) [inline specs/Endpoint/media.spec.php:42]: Maybe we prefer this format when we expect an exception?

```               
           expect(
                    [
                        $this->triggerCalendarReloadService,
                        'triggerCalendarReloadForProgram'
                    ]
                )
                    ->with($this->program_Golf->getId())
                    ->to->throw(Exception::class);
````

File: specs/Endpoint/media.spec.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        });
+        try {
+            $response = $this->mediaEndpoint->upload($imgUrl, [], $contentType);
+        } catch(Exception $e) {
+            expect($e)->to->be->instanceof(RuntimeException::class);
+        }
+        restore_error_handler();
+    });
+
+    it('should throw a Runtime Exception if the file is not found', function () {
```
Comment (rlehrke) [inline specs/Endpoint/media.spec.php:33]: context(

blocks are really ideal here

File: src/Endpoint/Media.php [hunk 4]
```diff
@@ -15,4 +18,47 @@ protected function getEndpoint()
...
     }
+
+    /**
+     * @param string $filePath absolute path of file to upload, or full URL of resource
+     * @param array $data
+     * @param string $mimeType if $filePath is a URL, the mime type of the file to be uploaded
+     * @throws \RuntimeException
+     * @return array
+     */
+    public function upload($filePath, $data = [], $mimeType = null)
```
Comment (rlehrke) [inline src/Endpoint/Media.php:29]: Do we like parameter types and return types here?

File: specs/Endpoint/media.spec.php (hunk 1 above)
Comment (boboudreau) [reply specs/Endpoint/media.spec.php:18]: I didn't want to actually hit up example.com every time the test ran, so, yeah, we "don't like this test right now".

File: src/Endpoint/Media.php (hunk 4 above)
Comment (boboudreau) [reply src/Endpoint/Media.php:29]: We can now! Forgot that we're dropping 5.6 so we can add in return types and scalar types on parameters.

Comment (boboudreau) [reply src/Endpoint/Media.php:29]: I won't `?string $mimeType` for now because nullable params weren't introduced until 7.1...

**Pull request merged**